### PR TITLE
Breaking out standalone dycore components

### PR DESCRIPTION
**Pull request by jkozdon**
This breaks the ODEsolver, mesh types, and MPI-aware arrays off of the AtmosDycore in the hopes that they will be reusable components

### CODE REVIEW

File: src/Arrays/MPIStateArrays.jl [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 using MPI
 
-export AtmosStateArray
+export MPIStateArray
 
 """
-    AtmosStateArray{S <: Tuple, T, DeviceArray, N,
-                    DATN<:AbstractArray{T,N}, Nm1, DAI1} <: AbstractArray{T, N}
+    MPIStateArray{S <: Tuple, T, DeviceArray, N,
+                  DATN<:AbstractArray{T,N}, Nm1, DAI1} <: AbstractArray{T, N}
```
Comment (simonbyrne) [inline src/Arrays/MPIStateArrays.jl]: Happy to keep this for the time being, but I was wondering if we could come up with a better name for this. Maybe `MPIGhostArray` or something like that?

Comment (jkozdon) [reply src/Arrays/MPIStateArrays.jl]: Not sure what the word `Ghost` means in this context. I tend to think of `Ghost` being data that an mpi rank depends another ran owns.

**Comment by jkozdon** [comment]
> Most of this looks like re-naming / restructuring, so this is fine by me.

Yes that's all it really is. Trying to pull stuff out of the AtmosDycore that is not AtmosDycore specific

File: src/Arrays/MPIStateArrays.jl (hunk 1 above)
Comment (jkozdon) [reply src/Arrays/MPIStateArrays.jl]: But FYI, I'm not wed to this name... just had to name it something other than `AtmosStateArray`. @lcw might have thoughts.

Comment (simonbyrne) [reply src/Arrays/MPIStateArrays.jl]: Let's leave it as is for now.

**Comment by simonbyrne** [comment]
Looks good! Just need to link the new markdown files into `docs/make.jl`

Would it make sense to move the `Dycore` directory to the top-level? (I realise this could also be done as part of your later work)

**Comment by jkozdon** [comment]
> Looks good! Just need to link the new markdown files into `docs/make.jl`

Great. Updated the commits to include this.

> Would it make sense to move the `Dycore` directory to the top-level? (I realise this could also be done as part of your later work)

Might be best to leave for now so as to minimize trouble for Akshay and Simone later this week?

Were you just thinking that it should live in something like `src/AtmosDycore` eventually?

**Comment by simonbyrne** [comment]
> Were you just thinking that it should live in something like `src/AtmosDycore` eventually?

Or just `src/Dycore` (since the hope is that we can share at least some of it with the ocean)

**Pull request merged**